components/ai_chat/resources/untrusted_conversation_frame/components/regenerate_answer_menu/regenerate_answer_menu.test.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import * as Mojom from '../../../common/mojom'
+import { render, waitFor, act } from '@testing-library/react'
+import { RegenerateAnswerMenu } from '.'
+import '@testing-library/jest-dom'
+
+const mockModels = [
+  {
+    key: '1',
+    displayName: 'Model One',
+    visionSupport: false,
+    options: {
+      leoModelOptions: {
+        name: 'model-one',
+        displayMaker: 'Company',
+        category: Mojom.ModelCategory.CHAT,
+        access: Mojom.ModelAccess.BASIC,
+        maxAssociatedContentLength: 10000,
+        longConversationWarningCharacterLimit: 9700
+      },
+      customModelOptions: undefined
+    }
+  },
+  {
+    key: '2',
+    displayName: 'Model Two',
+    visionSupport: true,
+    options: {
+      leoModelOptions: {
+        name: 'model-two-premium',
+        displayMaker: 'Company',
+        category: Mojom.ModelCategory.CHAT,
+        access: Mojom.ModelAccess.PREMIUM,
+        maxAssociatedContentLength: 10000,
+        longConversationWarningCharacterLimit: 9700
+      },
+      customModelOptions: undefined
+    }
+  }
+]
+
+describe('regenerate answer menu', () => {
+  it('ensures the menu opens and is visible', async () => {
+    render(
+      <RegenerateAnswerMenu
+        isOpen={false}
+        onOpen={() => {}}
+        onClose={() => {}}
+        onRegenerate={(selectedModelKey: string) => {
+          console.log('selectedModelKey', selectedModelKey)
+        }}
+        leoModels={mockModels}
+        turnModelKey='1'
+      />
+    )
+
+    // Make sure the anchor button is visible
+    const achorButton: any = document.querySelector('leo-button')!
+    expect(achorButton).toBeInTheDocument()
+    expect(achorButton).toBeVisible()
+
+    // Click the anchor button and make sure the menu opens
+    achorButton?.shadowRoot?.querySelector('button').click()
+    const menu: any = await waitFor(
+      () => document.querySelector('leo-buttonmenu')!
+    )
+    expect(menu).toBeInTheDocument()
+    expect(menu).toHaveTextContent('regenerateAnswerMenuTitle')
+    expect(menu).toHaveTextContent('regenerateAnswerButtonLabel')
+    expect(menu).toHaveTextContent('Model One')
+    expect(menu).toHaveTextContent('Model Two')
+  })
+
+  it('switches models and regenerates the answer', async () => {
+    const onRegenerateMock = jest.fn()
+    render(
+      <RegenerateAnswerMenu
+        isOpen={false}
+        onOpen={() => {}}
+        onClose={() => {}}
+        onRegenerate={onRegenerateMock}
+        leoModels={mockModels}
+        turnModelKey='1'
+      />
+    )
+
+    const achorButton: any = document.querySelector('leo-button')
+    achorButton?.shadowRoot?.querySelector('button').click()
```

Comment:
nit:
```suggestion
    const anchorButton: any = document.querySelector('leo-button')
    anchorButton?.shadowRoot?.querySelector('button').click()
```

Reply:
Fixed in https://github.com/brave/brave-core/pull/28999/commits/9a945bc196541d92e20722774d93b1536ae19bb8